kubernetes/kubernetes.go

```diff
@@ -186,6 +187,41 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 
 }
 
+func performConnectionChecks(ka *KubeAgentConfig) {
+	log.Info("Performing connectivity checks. Checking that the agent can retrieve S3 URL")
+
+	cldyMetricClient, err := client.NewHTTPMetricClient(client.Configuration{
+		Token:         ka.APIKey,
+		Verbose:       false,
+		ProxyURL:      ka.OutboundProxyURL,
+		ProxyAuth:     ka.OutboundProxyAuth,
+		ProxyInsecure: ka.OutboundProxyInsecure,
+	})
+	if err != nil {
+		log.Fatalf("error creating Cloudability Metric client: %v ", err)
+	}
+
+	metricSampleURL := apiEndpoint + "/metricsample"
+
+	file, err := os.Create("/tmp/temp.txt")
+	if err != nil {
+		log.Warnf("Failed to create file temp.txt")
```

Comment:
If you don't return from here, line 211 will panic. Each of these return should probably trigger a failure to start the agent as they're basic file io steps.

Reply:
Returning an error from here and logging it from the caller might be wise as well, would be easier to force an agent restart that way as well.

Reply:
good catch, making changes, thanks!